flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCleanerTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.concurrent.Executors;
+import org.apache.flink.runtime.state.testutils.TestCompletedCheckpointStorageLocation;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.concurrent.ExecutorService;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.flink.runtime.checkpoint.CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION;
+import static org.apache.flink.util.Preconditions.checkState;
+import static org.junit.Assert.assertTrue;
+
+public class CheckpointCleanerTest {
+
+    @Test
+    public void testTolerateFailureInPostCleanupSubmit() throws InterruptedException {
```

Comment:
I don't fully understand the test name. What are we testing here?

Reply:
It tests that failure to submit a post-cleanup callback don't propagate.

We probably need to adopt some spec-based test framework :)

Reply:
Ok, I think the problem is not the spec-based test framework but that I don't see where the failure from the submission of the post-cleanup callback is coming from. When I remove your changes of this PR, then the test also passes. Hence, this indicates to me that we are not testing the change of this PR here.

Reply:
I dropped this commit alltogether as a result of the above discussion.